datadriven/src/datadriven.rs

```diff
@@ -0,0 +1,217 @@
+use crate::test_data::TestData;
+use crate::test_data_reader::TestDataReader;
+use anyhow::Result;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+/// The main function to run tests
+///
+/// You need to pass the path of `testdata` where store the test cases, and your function
+/// to output expected result.
+///
+/// It will be compare with crate `difference`
+///
+/// It invokes a data-driven test. The test cases are contained in a
+/// separate test file and are dynamically loaded, parsed, and executed by this
+/// testing framework. By convention, test files are typically located in a
+/// sub-directory called "testdata". Each test file has the following format:
+///
+/// ```txt
+/// <command> [arg | arg=val | arg=(val1, val2, ...)]...
+/// <input to the command>
+/// ----
+/// <expected results>
+/// ````
+///
+/// The command input can contain blank lines. However, by default, the expected
+/// results cannot contain blank lines. This alternate syntax allows the use of
+/// blank lines:
+/// ```txt
+/// <command> [arg | arg=val | arg=(val1, val2, ...)]...
+/// <input to the command>
+/// ----
+/// ----
+/// <expected results>
+///
+/// <more expected results>
+/// ----
+/// ----
+/// ````
+/// Note: blank line is required after the second *`----`*.
+pub fn run_test<F>(path: &str, f: F) -> Result<()>
+where
+    F: FnOnce(&TestData) -> String + Copy,
+{
+    let entries: Vec<PathBuf> = fs::read_dir(path)?
```

Comment:
So test data has to be stored in a directory?

Reply:
yes, the same function is used in the directory, and run all of them.

Reply:
How about providing two functions, one for directory, one for single file? So users don't have to put test data into a directory.